coalib/parsing/FilterHelper.py

```diff
@@ -14,7 +15,21 @@ def is_valid_filter(filter):
     return filter in available_filters
 
 
-def apply_filter(filter_name, filter_args, all_bears=None):
+def apply_filter(filter_name, filter_args, all_bears=None, filter_function):
```

Comment:
The order of arguments in wrong. Default arguments are always present after non-default arguments. The correct order should be: `filter_name, filter_args, filter_function, all_bears=None`

Reply:
If we are changing the API, I think its better to keep it like this:
`filter_function, filter_name, filter_args, all_bears=None`.

@Makman2 Can we change this API ?

Reply:
If it was not part of a release yet it's okay :+1:

However, is `filter_function` necessary? This doesn't look very good and seems to defy the design^^

Reply:
Yup, not required. It should be derived from `filter_name`.